grant-project/grant-monorepo
# Email Scaffolding + Signup Email

Begins to address #143.

## What This Does

Sets up a precedent for sending emails via SendGrid, and implements the first one for when you sign up. It's a "Validate your email" email that doesn't actually function yet, but looks the part. Here are most of the changes:

* Added a `send_email` function to flask for sending emails.
    * Email settings are configured in a dict in `backend/grant/email/send.py`
    * Email templates are done in `backend/grant/templates/email/{type}.html|txt`
    * Emails can be designed pretty easily using Litmus, hit me up for login creds and if you need more info.
* Added instructions to `.env.example` and `backend/README.md` to get emails working. Development functions perfectly fine without setup, you'll just get errors in your log.
* Adds an email that sends every time you POST a new user.

The email template doesn't look perfect yet, but we can easily change it and have all emails inherit the style changes.

## Steps to Test

* Follow the instructions in `backend/README.md` and add the new environment variable.
* Sign up as a new user.
* Make sure the email comes in & renders properly.

## Screenshots

<img width="1440" alt="screen shot 2018-10-18 at 7 13 05 pm" src="https://user-images.githubusercontent.com/649992/47189942-fb100700-d30c-11e8-8be5-b97ed8634cea.png">
